contrib/document_constants.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+
+#
+# (C) 2021 ProFUSION Sistemas e Solucoes LTDA
+#
+# All files and artifacts in this repository are licensed under the
+# provisions of the license provided by the LICENSE file in this repository.
+#
+
+import argparse
+import sys
+from enum import Enum
+from textwrap import dedent
+from typing import Mapping, Sequence, Type, Union
+
+import vspec.model.constants as constants
+
+
+# NOTE: only used to fetch the auto-generated __doc__
+class GenericEnum(Enum):
+    placeholder = 1
+
+
+def create_table(
+    header: Sequence[str],
+    rows: Sequence[Sequence[str]],
+    size_rounding: int = 4,
+):
+    """Generate markdown table
+
+    It will measure all the row cells and their header to generate
+    a human friendly table (the text representation looks like a table).
+
+    To avoid too much resizing when the cells change, the sizes are rounded
+    using `size_rounding`.
+    """
+    def round(n):
+        remainder = n % size_rounding
+        if not remainder:
+            return n
+        return n + remainder
```

Comment:
I didn't do much thinking on this one, I just wanted to have some way to avoid resizing, we can indeed change it to be as you said, it would make more sense, indeed.

Reply:
fixed in the updated version, also with doctests :-)